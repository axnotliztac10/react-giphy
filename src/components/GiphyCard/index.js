import React from 'react';
import { Image, Button, Text } from 'rebass';

import Card from 'components/Card';
import addButton from 'images/add.png';
import addedButton from 'images/added.png';
import loader from 'images/loader.svg';

export default class GiphyCard extends React.Component {
  state = {
    loaded: false
  }

  onLoad = () => {
    this.setState({
      loaded: true
    });
  }

  render() {
    const { loaded } = this.state;
    const { giphy, onAddRemoveFavorite } = this.props;

    const Img = () => (
      <img
        alt={giphy.title}
        style={{
          maxWidth: '100%',
          display: (loaded) ? 'block' : 'none'
        }}
        src={giphy.images.downsized_medium.url}
        onLoad={this.onLoad}
      />
    );

    return (
      <Card
        p={2}
        my={3}
        mx={2}
        width={[1, 1/3, 1/4]}
      >
        <Img />
        {!loaded && <img src={loader} alt="loading" />}
        <Button
          p={0}
          width={1}
          variant="outline"
          bg="transparent"
          onClick={() => onAddRemoveFavorite(giphy)}
        >
          <Text pb={2} color="black">{giphy.title}</Text>
          {!giphy.isFavorite && <Image src={addButton} width={1/4} />}
          {giphy.isFavorite && <Image src={addedButton} width={1/4} />}
        </Button>
      </Card>
    );
  }
}
