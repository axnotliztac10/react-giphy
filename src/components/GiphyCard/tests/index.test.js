import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Button } from 'rebass';

import GiphyCard from '../index';

describe('<Card />', () => {
  const giphy = {
    title: '',
    images: {
      downsizedMedium: { url: '' }
    }
  };
  const wrapper = shallow(
    <GiphyCard
      giphy={giphy}
      onAddRemoveFavorite={() => {}}
    />
  );

  it('should render correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Should trigger action', () => {
    wrapper.find(Button).simulate('click');
  });
});
