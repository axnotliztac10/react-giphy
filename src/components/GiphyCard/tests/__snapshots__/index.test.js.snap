// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correctly 1`] = `
<Card
  mx={2}
  my={3}
  p={2}
  width={
    Array [
      1,
      0.3333333333333333,
      0.25,
    ]
  }
>
  <Img />
  <img
    alt="loading"
    src="loader.svg"
  />
  <Styled(styled.div)
    as="button"
    bg="transparent"
    border={0}
    borderRadius={4}
    color="white"
    fontSize="inherit"
    fontWeight="bold"
    m={0}
    onClick={[Function]}
    p={0}
    px={3}
    py={2}
    variant="outline"
    width={1}
  >
    <Styled(styled.div)
      color="black"
      pb={2}
    />
    <Styled(styled.div)
      as="img"
      m={0}
      src="add.png"
      width={0.25}
    />
  </Styled(styled.div)>
</Card>
`;
